src/rmap/core.clj

```diff
@@ -64,7 +64,8 @@
   not evaluated yet. The body can use the [[ref]] function while it is
   evaluated, or you can bind it locally for use at a later stage."
   [& body]
-  `(RVal. (fn [ref#]
+  `(RVal. ^{:ref/sexp '~@body}
```

Comment:
The unquote-splice would make the macro fail if the body contains multiple expressions like `(rval (println "lazy!") "hi")`.